client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncCollection.java

```diff
@@ -0,0 +1,161 @@
+import com.tigrisdata.db.client.model.DeleteResponse;
+import com.tigrisdata.db.client.model.Field;
+import com.tigrisdata.db.client.model.InsertOrReplaceRequestOptions;
+import com.tigrisdata.db.client.model.InsertOrReplaceResponse;
+import com.tigrisdata.db.client.model.InsertRequestOptions;
+import com.tigrisdata.db.client.model.InsertResponse;
+import com.tigrisdata.db.client.model.ReadRequestOptions;
+import com.tigrisdata.db.client.model.TigrisCollectionType;
+import com.tigrisdata.db.client.model.TigrisFilter;
+import com.tigrisdata.db.client.model.UpdateRequestOptions;
+import com.tigrisdata.db.client.model.UpdateResponse;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public interface TigrisAsyncCollection<T extends TigrisCollectionType> {
+
+  /**
+   * @param filter filter to narrow down read
+   * @param fields optionally specify fields you want to be returned from server
+   * @param readRequestOptions read options
+   * @param reader reader callback
+   * @throws TigrisDBException
+   */
+  void read(
+      TigrisFilter filter,
+      List<Field<?>> fields,
+      ReadRequestOptions readRequestOptions,
+      TigrisDBAsyncReader<T> reader)
+      throws TigrisDBException;
+
+  /**
+   * Reads matching documents
+   *
+   * @param filter filter to narrow down read
+   * @param fields optionally specify fields you want to be returned from server
+   * @param reader reader callback
+   * @throws TigrisDBException in case of an error
+   */
+  void read(TigrisFilter filter, List<Field<?>> fields, TigrisDBAsyncReader<T> reader)
+      throws TigrisDBException;
+
+  /**
+   * Reads a single document. This method is generally recommended for point lookup, if used for
+   * non-point lookup any arbitrary matching document will be returned.
+   *
+   * @param filter
+   * @return a future to the document
+   * @throws TigrisDBException
+   */
+  CompletableFuture<T> readOne(TigrisFilter filter) throws TigrisDBException;
+
+  /**
+   * @param documents list of documents to insert
+   * @param insertRequestOptions insert option
+   * @return a future to the {@link InsertResponse}
+   * @throws TigrisDBException in case of an error
+   */
+  CompletableFuture<InsertResponse> insert(
+      List<T> documents, InsertRequestOptions insertRequestOptions) throws TigrisDBException;
+
+  /**
+   * @param documents list of documents to insert
+   * @return a future to the {@link InsertResponse}
+   * @throws TigrisDBException in case of an error
+   */
+  CompletableFuture<InsertResponse> insert(List<T> documents) throws TigrisDBException;
+
+  /**
+   * inserts a single document to the collection
+   *
+   * @param document
+   * @return a future to the {@link InsertResponse}
+   * @throws TigrisDBException
+   */
+  CompletableFuture<InsertResponse> insert(T document) throws TigrisDBException;
+
+  /**
+   * Inserts the documents if they don't exist already (matched by primary key), replaces them
```

Comment:
remove `(matched by primary key)` as it applies to all write apis so not adding any benefit adding here.

Reply:
removed from all docs.